labs/b6.md

```diff
@@ -0,0 +1,144 @@
+
+Now go through the same process again. This time, send the parent a `SIGHUP` (hangup) signal. Can you still find the sleep process? When `SIGHUP` is sent to a parent shell, the parent subsequently sends hangup signals to any child processes before terminating; all processes that receive `SIGHUP` from a parent shell will terminate -- this is one way to avoid creating orphan processes.
+
+If you are interested in learning about the different signals, run `man 7 signal`. 
+
+## Cron
+So much infrastructure in the computing world relies on scheduled processes. This is the job of the cron daemon. For example, at the OCF, we use a cron job that runs every thirty minutes to keep our servers and desktops updated via puppet. To get a feel for the cron scheduler, we're going to write a basic cron job.
+
+Open the cron editor by running `crontab -e` (if the editor of your choice isn't being launched, set the `EDITOR` environment variable), which will create a crontab for your user. Here is a sample task:
+
+```* * * * * date +"\%T" >> $HOME/timestamps.txt```
+
+Right now this runs every minute. Modify it to run every five minutes. Make sure to leave this running while you complete the remainder of the lab as you will be turning this file in.
+
+If you get stuck, visit crontab.guru!
+
+## Using systemd
+
+### What services are running right now? 
+Run `systemctl`. You’ll see a long table of every unit known to systemd.
+
+Let’s narrow it down to services for now. Run `systemctl --type=service`. Now you can see a list of all services running on your computer. Each of these services is a daemon running in the background. Do you see any familiar services running?
+
+### Controlling Services 
+Now let's use `systemd` to control a an nginx web server. Install nginx by issuing `sudo apt install nginx`. Once that is done we can tell systemd to start the service with the following: `sudo systemctl start nginx`. Run `systemctl status nginx` to ensure it is running and navigate to http://*yourvm*.decal.xcf.sh/ -- you should be greeted by the nginx default landing page.
+
+Now let's make nginx listen for connections on the nonstandard port 420. In `/etc/nginx/sites-available/default` change the following lines:
+
+```nginx
+listen 80 default_server;
+listen [::]:80 default_server;
+```
+
+to:
+```nginx
+listen 420 default_server;
+listen [::]:420 default_server;
+```
+
+Tell systemd that nginx has changed configuration and needs reloading with: `sudo systemctl reload nginx`. Now, accessing  http://*yourvm*.decal.xcf.sh/ should now give you a connection refused error and your webserver will only be accessible via http://*yourvm*.decal.xcf.sh:420/.
+
+Note that not all services can be reloaded; systemd will notify you if this is the case and such services will have to be restarted instead.
+
+Finally go ahead and stop the nginx service. What is the command to do this? 
+
+### Creating a service
```

Comment:
One possibility for the beginner track is to give them some starter code instead and fill in the most interesting fields.

Reply:
Yeah. I think a file with all the necessary fields laid out is good. Then students just fill in the `*` in the `Something=*`, and `WantedBy=multi-user.target` will be given since we shouldn't presume knowledge of run levels.